addons/project/project.py

```diff
@@ -896,13 +896,22 @@ def message_new(self, cr, uid, msg, custom_values=None, context=None):
             'planned_hours': 0.0,
             'partner_id': msg.get('author_id', False)
         }
+        Partner = self.pool['res.partner']
         defaults.update(custom_values)
         res = super(task, self).message_new(cr, uid, msg, custom_values=defaults, context=context)
-        email_list = tools.email_split((msg.get('to') or '') + ',' + (msg.get('cc') or ''))
+        email_data = ('%s,%s') % (msg.get('cc'), msg.get('from'))
+        email_list = tools.email_split(email_data)
         new_task = self.browse(cr, uid, res, context=context)
         if new_task.project_id and new_task.project_id.alias_name:  # check left-part is not already an alias
             email_list = filter(lambda x: x.split('@')[0] != new_task.project_id.alias_name, email_list)
-        partner_ids = filter(lambda x: x, self._find_partner_from_emails(cr, uid, email_list, check_followers=False))
+        partner_ids = []
+        for email in email_list:
+            partner_id = filter(lambda x: x, self._find_partner_from_emails(cr, uid, [email], check_followers=False))
+            if not partner_id:
+                for name_email in email_data.split(','):
```

Comment:
may be simple logic:
1) prepare dict like {'email': name_email} from email_data
1) prepare todo email list from email list which does not link with any partner.
2) create partner with full name (name_email) for all todo email list